### PR TITLE
[SYCL][ESIMD] Enable tests passing with newer GPU driver

### PR DESCRIPTION
**Pull request by turinevgeny**
Closes intel/llvm#11358.

### CODE REVIEW

**Review by sarnex — APPROVED** [review]

File: sycl/test-e2e/ESIMD/assert.cpp [hunk 1]
```diff
@@ -2,11 +2,6 @@
 
 // RUN: %{build} -DSYCL_FALLBACK_ASSERT=1 -o %t.out
 // RUN: %{run} %t.out &> %t.txt ; FileCheck %s --input-file %t.txt
-//
-// The test still fails after GPU driver update on Linux. Temporary marking it
-// as expected to fail, whilst it is being investigated, see intel/llvm#11359
```
Comment (v-klochkov) [inline sycl/test-e2e/ESIMD/assert.cpp:7]: I did not understand what happened. Did you?
It seems the GPU driver was updated and it caused regression. Then this test was disabled. But not it reports pass with same updated GPU driver.
Is the issue really fixed, or the test is flaky?

If the issue mentioned in this comment is really fixed, please add a corresponding comment in the description of the patch. Something like 'closes #11359'

Comment (turinevgeny) [reply sycl/test-e2e/ESIMD/assert.cpp:7]: It is passing on Gen12, and is flaky on PVC. I updated REQUIRES line.

File: sycl/test-e2e/ESIMD/slm_block_load_store.cpp [hunk 2]
```diff
@@ -107,8 +102,8 @@ int main() {
   // which has been fixed only in "1.3.26816", and in win/opencl version going
   // _after_ 101.4575.
   if (!esimd_test::isGPUDriverGE(Q, esimd_test::GPUDriverOS::LinuxAndWindows,
-                                 "26816", "101.4576")) {
-    std::cout << "Skipped. The test requires GPU driver 1.3.26816 or newer.\n";
+                                 "26690", "101.4827")) {
```
Comment (v-klochkov) [inline sycl/test-e2e/ESIMD/slm_block_load_store.cpp:105]: That is odd that the test passes with older driver(26690) than the previously requested one(26816), but if that really passes, then fine. 

nit: The comment above is not consistent with the new changes/driver-checks.

File: sycl/test-e2e/ESIMD/assert.cpp [hunk 3]
```diff
@@ -1,12 +1,10 @@
-// REQUIRES: linux && level_zero
+// TODO: This test is flaky on PVC and stable on gen12. Investigate PVC fail
+// and remove Gen12 limitation.
+// see intel/llvm#11359
+// REQUIRES: linux, level_zero, gpu-intel-gen12
```
Comment (aelovikov-intel) [inline sycl/test-e2e/ESIMD/assert.cpp:4]: We don't set `gpu-intel-gen12` in the upstream CI. I'm working on enabling ATS-M run in post-commit, but even that would use `gpu-intel-dg2`.

**Pull request closed without merging**